### PR TITLE
Expose generate_alignment + generate_hgvs_variants_from_alignment functions

### PR DESCRIPTION
**Pull request by austin-mbs**
## Description

As part of the alignment work, we are exposing some additional functions that can be used externally.
- generate_alignment (Builds a HGVS Alignment object)
- generate_hgvs_variants_from_alignment (Skipping alignment generation)

Also, as part of this PR
- Updated docs - make import rules more explicitly mentioned - internals should not be used

## Relevant Links

N/A

### Screenshots & Media

N/A

### Testing

- Unit Tests added

### Code Flow

N/A

### Edge cases / Breaking Changes / Known Issues

N/A

### CODE REVIEW

File: README.md
Comment (heuermh) [inline README.md:1]: Minor typo, provides &rarr; provide

File: palamedes/__init__.py
Comment (heuermh) [inline palamedes/__init__.py:1]: boolean flags for strategies are not my favorite, could a separate method make a better API?  Or perhaps allow the strategy to be passed in?

Comment (mbiokyle29) [reply palamedes/__init__.py:1]: This is existing functionality that is just getting moved so it would be a breaking change to get rid of the flag. Don't think it's worth doing right now, but I am open to a change in a follow up PR if you feel its important. Personally I find the ergonomics of passing a boolean to be a bit better then some string that represents the "method" you want (we'd likely want some kind of constant / enum, etc). If there were a bunch of methods it would make more sense.

File: palamedes/__init__.py [hunk 1]
```diff
@@ -1,16 +1,152 @@
...
 __version__ = "0.0.8"
 
 
+def generate_hgvs_variants_from_alignment(
+    alignment: Alignment, use_non_standard_substitution_rules: bool = False, molecule_type: str = MOLECULE_TYPE_PROTEIN
+) -> list[SequenceVariant]:
+    """
+    Given a pairwise alignment object and a molecule type, generate a list of HGVS SequenceVariants.
+
+    - Alignment: Generated via BioPython.PairwiseAligner - (`generate_alignment` for more information)
```
Comment (mbiokyle29) [inline palamedes/__init__.py:28]: "See `generate_alignemt` for more information"

File: palamedes/__init__.py [hunk 2]
```diff
@@ -1,16 +1,152 @@
...
+
+    - molecule_type: Currently only molecule type 'protein' is supported.
+
+    - An optional flag: `use_non_standard_substitution_rules` is a boolean flag which will enable logic that treats
+    multiple consecutive mismatches as separate subsitutions, vs merging together into a delins. This is against HGVS
+    spec but has utility for some use cases.
+
+
```
Comment (mbiokyle29) [inline palamedes/__init__.py:36]: Can you trim this excess white space?

File: palamedes/__init__.py [hunk 3]
```diff
@@ -1,16 +1,152 @@
...
+     REF_SEQUENCE_ID,
+)
 
 __version__ = "0.0.8"
 
 
+def generate_hgvs_variants_from_alignment(
+    alignment: Alignment, use_non_standard_substitution_rules: bool = False, molecule_type: str = MOLECULE_TYPE_PROTEIN
+) -> list[SequenceVariant]:
+    """
```
Comment (mbiokyle29) [inline palamedes/__init__.py:25]: Adding some examples here using the follow format might be nice for the API docs:
```
    .. code-block:: python

        >>> from palamedes import generate_hgvs_variants
        >>> generate_hgvs_variants("PFKISIHL", "TPFKISIH")
        [
            SequenceVariant(ac=ref, type=p, posedit=Pro1extThr-1, gene=None),
            SequenceVariant(ac=ref, type=p, posedit=Leu8del, gene=None),
        ]
```
This is from the doc string for `generate_hgvs_variants`

File: palamedes/__init__.py [hunk 4]
```diff
@@ -1,16 +1,152 @@
...
+    ]
+
+
+def generate_alignment(
+    reference_seq_record: SeqRecord,
+    alternate_seq_record: SeqRecord,
+    molecule_type: str = MOLECULE_TYPE_PROTEIN,
+    aligner: PairwiseAligner | None = None,
+) -> Alignment:
+    """
```
Comment (mbiokyle29) [inline palamedes/__init__.py:61]: Since this is public now, can you add some code example to the doc string, using the same block as above?

File: palamedes/__init__.py [hunk 5]
```diff
@@ -70,7 +206,9 @@ def generate_hgvs_variants(
     """
 
     if molecule_type not in BUILDER_CONFIG:
-        raise NotImplementedError(f"No HGVS builder is defined for molecule_type: {molecule_type}!")
+        raise NotImplementedError(
+            f"Type {molecule_type} unsupported or unrecognized! Current supported types: {','.join(BUILDER_CONFIG.keys())}"
+        )
```
Comment (mbiokyle29) [inline palamedes/__init__.py:211]: Thank you for making this error message better!

File: palamedes/test_palamedes.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
+from palamedes import generate_variant_blocks, generate_alignment
```
Comment (mbiokyle29) [inline palamedes/test_palamedes.py:1]: `generate_variant_blocks` is still defined in `align.py`, so the tests for it should stay in there. Only `generate_alignment` was moved.

**Review by mbiokyle29 — CHANGES_REQUESTED** [review]
Left a few notes on doc-strings and tests that should not have been moved. If you could also get tests added for the new `generate_hgvs_variants_from_alignment` and populate the PR description with a bit more info, that would be great. Thanks for this!

**Review by mbiokyle29 — APPROVED** [review]
LGTM, thanks!

**Review by mbiokyle29 — APPROVED** [review]
